terminus-ui/chart/src/chart.component.ts

```diff
@@ -173,9 +183,12 @@ export class TsChartComponent implements OnInit, OnChanges, OnDestroy {
   /**
    * Initialize a chart
    */
-  private init(type: TsChartVisualizationOptions): void {
+  private createChart(type: TsChartVisualizationOptions): void {
     this.zone.runOutsideAngular(() => {
-      // Create the appropriate chart using a chained ternary
+      if (this.chart) {
+        this.chart.dispose();
```

Comment:
Does this ensure that old chart gets destroyed before creating new one?

Reply:
yeah, if the chart exists, we call the chart's dispose method.